src/Jobs/SamlSso.php

```diff
@@ -203,9 +208,11 @@ private function getQueryParams()
 
     public function setSpCertificate()
     {
-        $this->sp_certificate = config(sprintf(
+        $filename = config(sprintf(
             'samlidp.sp.%s.certificate',
             $this->getServiceProvider($this->authn_request)
-        ));
+        ), 'cert.pem');
+
+        $this->certificate = (new X509Certificate)->loadPem(Storage::disk('samlidp')->get($filename));
```

Comment:
This is the place where the new certificate should be loaded. 
There's no reason to use a `$this->sp_certificate` variable when we can directly load the certificate into `$this->certificate`.

Reply:
`$this->certificate` is used for loading the Identity Provider certificate and you are trying to load the Service Provider using the same prop.